analysis/dataset_definition_dx_to_med_time.py

```diff
@@ -0,0 +1,93 @@
+from ehrql import create_dataset
+from ehrql.tables.tpp import (
+    patients,
+    practice_registrations,
+    clinical_events,
+    medications,
+)
+
+from codelists import (
+    adhd_codelist,
+    adhdrem_codelist, 
+    methylphenidate_codelist
+)
+
+def last_matching_event(events, codelist, where=True):
+    """Select the last matching SNOMED CT event from specified codelist
+
+    Args:
+        events (e.g., clinical_events or ): Many rows per patient event frame to select last matching event from codelist from
+        codelist (codelist): Clinical codelist, must be using snomedct codes
+        where (bool, optional): _description_. Defaults to True.
+
+    Returns:
+        patient frame: One row per patient frame, with the last matching event from codelist
+    """
+    return (
+        events.where(where)
+        .where(events.snomedct_code.is_in(codelist))
+        .sort_by(events.date)
+        .last_for_patient()
+    )
+
+dataset = create_dataset()
+dataset.configure_dummy_data(population_size=5)
+
+# Date range
+start_date = "2016-04-01"
+end_date = "2017-03-31"
```

Comment:
I think we need to remove the start and end dates for selecting the events for this analysis to make sure we pick up everything